src/particle_filter/include/particle_filter/distanceTransform.h

```diff
@@ -0,0 +1,177 @@
+  bound[1] = std::numeric_limits<double>::infinity();
+  double s;
+  if (dir == 0) {
+    for (int q = 1; q < range[dir]; q++) {
+      s = ((cost_fun[q][idx1][idx2] + double(q * q)) 
+	   - (cost_fun[envelope[k]][idx1][idx2] 
+	      + double(envelope[k] * envelope[k]))) 
+	/ double(2 * q - 2 * envelope[k]);
+      while (s <= bound[k])
+	{
+	  k = k - 1;
+	  s = ((cost_fun[q][idx1][idx2] + double(q * q)) 
+	       - (cost_fun[envelope[k]][idx1][idx2] 
+		  + double(envelope[k] * envelope[k]))) 
+	    / double(2 * q - 2 * envelope[k]);
+	}
+      k = k + 1;
+      envelope[k] = q;
+      bound[k] = s;
+      bound[k + 1] = std::numeric_limits<double>::infinity();
+    }
+
+    k = 0;
+    for (int q = 0; q < range[dir]; q++) {
+      while (bound[k + 1] < q) {
+	k = k + 1;
+      }
+      dist_transform_1D[q][idx1][idx2] = double((q - envelope[k]) * (q - envelope[k])) + cost_fun[envelope[k]][idx1][idx2];
+    }
+  }
+  else if (dir == 1) {
+    for (int q = 1; q < range[dir]; q++) {
+      s = ((cost_fun[idx1][q][idx2] + double(q * q)) 
+	   - (cost_fun[idx1][envelope[k]][idx2] 
+	      + double(envelope[k] * envelope[k]))) 
+	/ double(2 * q - 2 * envelope[k]);
+      while (s <= bound[k])
+	{
+	  k = k - 1;
+	  s = ((cost_fun[idx1][q][idx2] + double(q * q)) 
+	       - (cost_fun[idx1][envelope[k]][idx2] 
+		  + double(envelope[k] * envelope[k]))) 
+	    / double(2 * q - 2 * envelope[k]);
+	}
+      k = k + 1;
+      envelope[k] = q;
+      bound[k] = s;
+      bound[k + 1] = std::numeric_limits<double>::infinity();
+    }
+
+    k = 0;
+    for (int q = 0; q < range[dir]; q++) {
+      while (bound[k + 1] < q) {
+	k = k + 1;
+      }
+      dist_transform_1D[idx1][q][idx2] = 
+	double((q - envelope[k]) * (q - envelope[k]))
+	+ cost_fun[idx1][envelope[k]][idx2];
+    }
+  }
+  else {
+    for (int q = 1; q < range[dir]; q++) {
+      double t = (cost_fun[idx1][idx2][q] + double(q * q)) - 
+	(cost_fun[idx1][idx2][envelope[k]]);
+      s = ((cost_fun[idx1][idx2][q] + double(q * q)) - 
+	   (cost_fun[idx1][idx2][envelope[k]] + 
+	    double(envelope[k] * envelope[k]))) / double(2 * q - 2 * envelope[k]);
+      while (s <= bound[k])
+	{
+	  k = k - 1;
+	  s = ((cost_fun[idx1][idx2][q] + double(q * q)) - 
+	       (cost_fun[idx1][idx2][envelope[k]] + 
+		double(envelope[k] * envelope[k]))) / double(2 * q - 2 * envelope[k]);
+	}
+      k = k + 1;
+      envelope[k] = q;
+      bound[k] = s;
+      bound[k + 1] = std::numeric_limits<double>::infinity();
+    }
+
+    k = 0;
+    for (int q = 0; q < range[dir]; q++) {
+      while (bound[k + 1] < q) {
+	k = k + 1;
+      }
+      dist_transform_1D[idx1][idx2][q] = double((q - envelope[k]) * (q - envelope[k])) + cost_fun[idx1][idx2][envelope[k]];
+    }
+  }
+  delete[] envelope;
+  delete[] bound;
+}
+
+void distanceTransform::build(double cube_para[3], double World_Range[3][2])
+{
+  const double LARGE_NUM = 10000000;
+  double x, y, z;
+  memcpy(world_range, World_Range, 6 * sizeof(double));
```

Comment:
Your use of capital variable and lower case variables is inconsistent. One possibility, end member variables with _
memcpy(world_range_, world_range,...)

If you use std::vectors you don't need memcpy